packages/front-end/components/Settings/Team/SingleRoleSelector.tsx

```diff
@@ -67,7 +67,28 @@ export default function SingleRoleSelector({
     groupedOptions.push({ label: "Custom", options: customOptions });
   }
 
-  const availableEnvs = useEnvironments();
+  const activeEnvs = useEnvironments();
+
+  // Create a list of envs that combines current active envs, and any deleted envs that are still in the value.environments
+  const allEnvs = useMemo(() => {
+    const activeEnvIds = new Set(activeEnvs.map((env) => env.id));
+    const envOptions = [...activeEnvs];
```

Comment:
Nit: type this as `SingleValue[]`.  Then, you can just pass it directly into the select field below and keep all of the logic contained within this memo.